### PR TITLE
[12.0] [MIG] partner_phonecall_schedule

### PR DESCRIPTION
**Pull request by anandkansagra**
Issue [Reference](https://github.com/OCA/partner-contact/issues/630)

### CODE REVIEW

File: partner_phonecall_schedule/models/res_partner.py [hunk 1]
```diff
@@ -30,12 +32,12 @@ class ResPartner(models.Model):
     @api.depends("phonecall_calendar_ids", "phonecall_calendar_attendance_ids")
     def _compute_phonecall_available(self):
         """Know if a partner is available to call right now."""
-        Attendance = self.env["resource.calendar.attendance"]
+        attendance = self.env["resource.calendar.attendance"]
```
Comment (pedrobaeza) [inline partner_phonecall_schedule/models/res_partner.py]: This is upper on purpose, for knowing that this is the model handler. This is done the same on Odoo.

File: partner_phonecall_schedule/models/res_partner.py [hunk 2]
```diff
@@ -47,24 +49,20 @@ def _compute_phonecall_calendar_ids(self):
 
     def _search_phonecall_available(self, operator, value):
         """Search quickly if partner is available to call right now."""
-        Attendance = self.env["resource.calendar.attendance"]
-        available = Attendance.search(
+        attendance = self.env["resource.calendar.attendance"]
```
Comment (pedrobaeza) [inline partner_phonecall_schedule/models/res_partner.py]: The same

File: partner_phonecall_schedule/models/res_partner.py [hunk 3]
```diff
@@ -47,24 +49,20 @@ def _compute_phonecall_calendar_ids(self):
...
 
     def _phonecall_available_domain(self):
         """Get a domain to know if we are available to call a partner."""
-        now = self.env.context.get("now", datetime.now())
-        try:
-            now = fields.Datetime.from_string(now)
-        except TypeError:
-            # `now` is already a datetime object
-            pass
+        now = fields.Datetime.from_string(self.env.context.get(
```
Comment (pedrobaeza) [inline partner_phonecall_schedule/models/res_partner.py:64]: Use directly `datetime` Python type, and use `fields.Datetime.now()`.

File: partner_phonecall_schedule/tests/test_schedule.py [hunk 4]
```diff
@@ -13,9 +13,9 @@ class CanICallCase(SavepointCase):
     @classmethod
     def setUpClass(cls):
         super(CanICallCase, cls).setUpClass()
-        cls.Calendar = cls.env["resource.calendar"].with_context(tz="UTC")
-        cls.Partner = cls.env["res.partner"].with_context(tz="UTC")
-        cls.some_mornings = cls.Calendar.create({
+        cls.calendar = cls.env["resource.calendar"].with_context(tz="UTC")
```
Comment (pedrobaeza) [inline partner_phonecall_schedule/tests/test_schedule.py]: Undo this.

File: partner_phonecall_schedule/tests/test_schedule.py [hunk 5]
```diff
@@ -64,24 +64,24 @@ def setUp(self):
         self.datetime = datetime(2017, 9, 15, 10, 53, 30)
 
     def _allowed(self, now=None):
-        dude, Partner = self.dude, self.Partner
+        dude, partner = self.dude, self.partner
         if now:
             dude = dude.with_context(now=now)
-            Partner = Partner.with_context(now=now)
+            partner = partner.with_context(now=now)
```
Comment (pedrobaeza) [inline partner_phonecall_schedule/tests/test_schedule.py]: Undo this.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
@anandkansagra most of your changes shouldn't be done, as it follows a different style of what you are used to, but valid, and in a migration, you should reduce the diff at maximum although it doesn't match your style. It's when doing refactoring when you can change this style if wanted.

File: partner_phonecall_schedule/models/res_partner.py (hunk 3 above)
Comment (anandkansagra) [reply partner_phonecall_schedule/models/res_partner.py:64]: If we will use `fields.Datetime.now()` then UT will be failed due to  `self.assertTrue(dude.phonecall_available)`

Comment (anandkansagra) [reply partner_phonecall_schedule/models/res_partner.py:64]: 2018-12-25 04:53:34,633 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: FAIL: test_friday_morning (odoo.addons.partner_phonecall_schedule.tests.test_schedule.CanICallCase) 
2018-12-25 04:53:34,634 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: ` I can call dude this morning 
2018-12-25 04:53:34,634 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: Traceback (most recent call last): 
2018-12-25 04:53:34,634 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: `   File "/home/anand/workspace/oca/anand/partner-contact/partner_phonecall_schedule/tests/test_schedule.py", line 138, in test_friday_morning 
2018-12-25 04:53:34,634 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: `     self.allowed() 
2018-12-25 04:53:34,634 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: `   File "/home/anand/workspace/oca/anand/partner-contact/partner_phonecall_schedule/tests/test_schedule.py", line 94, in allowed 
2018-12-25 04:53:34,634 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: `     self._allowed() 
2018-12-25 04:53:34,634 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: `   File "/home/anand/workspace/oca/anand/partner-contact/partner_phonecall_schedule/tests/test_schedule.py", line 71, in _allowed 
2018-12-25 04:53:34,634 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: `     self.assertTrue(dude.phonecall_available) 
2018-12-25 04:53:34,634 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: ` AssertionError: False is not true

Comment (pedrobaeza) [reply partner_phonecall_schedule/models/res_partner.py:64]: OK

**Review by tbaden — APPROVED** [review]
code review lgtm
and @pedrobaeza changes were done or am I missing one?

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**